File: OMOPExpressAPI/controllers/metadataController.js
const pool = require('../db');

exports.getAllMetadata = async (req, res) => {
  try {
    const { rows } = await pool.query('SELECT * FROM metadata LIMIT 100');
    res.json(rows);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

exports.getMetadataById = async (req, res) => {
  try {
    const { id } = req.params;
    const { rows } = await pool.query('SELECT * FROM metadata WHERE metadata_id = $1', [id]);
    if (rows.length === 0) return res.status(404).json({ error: 'Not found' });
    res.json(rows[0]);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

File: OMOPRubyAPI/omoprubyapi/app/controllers/api/metadata_controller.rb
module Api
  class MetadataController < ApplicationController
    def index
      render json: Metadata.limit(100)
    end

    def show
      m = Metadata.find_by(metadata_id: params[:id])
      m ? render(json: m) : render(json: { error: 'Not found' }, status: :not_found)
    end
  end
end
